src/Sportalytics.Feed.Persistence.MongoDB/Core/MongoRepository.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Linq.Expressions;
+using MongoDB.Driver;
+using Sportalytics.Feed.Domain.Interfaces;
+using Sportalytics.Feed.Persistence.MongoDB.Interfaces;
+
+namespace Sportalytics.Feed.Persistence.MongoDB.Core;
+
+public class MongoRepository<T>(IMongoDatabase database, string collectionName) : IRepository<T> where T : IEntity
+{
+    private readonly IMongoCollection<T> _collection = database.GetCollection<T>(collectionName);
+    private readonly FilterDefinitionBuilder<T> _filterDefinitionBuilder = Builders<T>.Filter;
+
+    public IQueryable<T> Query(Expression<Func<T, bool>> filter)
+    {
+        var foundList = _collection.AsQueryable().Where(filter);
+        return foundList;
+    }
+
+    public async Task AddAsync(T entity, CancellationToken cancellationToken)
+    {
+        await _collection.InsertOneAsync(entity, options: null, cancellationToken);
+    }
+
+    public async Task UpdateAsync(T entity, CancellationToken cancellationToken)
+    {
+        var filter = _filterDefinitionBuilder.Eq(e => e.Id, entity.Id);
+        await _collection.ReplaceOneAsync(filter, entity, cancellationToken: cancellationToken);
```

Comment:
Will it be full replace? Or only for updated fields?

Reply:
Yep. According documentation: 
`You can perform a replace operation in MongoDB with the ReplaceOne() method. This method removes all fields (except the _id field) from the first document that matches the search criteria, then inserts the fields and values you specify into the document.`
it will be replaced fully.